x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/InferenceNamedWriteablesProvider.java

```diff
@@ -184,6 +189,9 @@ public static List<NamedWriteableRegistry.Entry> getNamedWriteables() {
         namedWriteables.add(
             new NamedWriteableRegistry.Entry(TaskSettings.class, CohereEmbeddingsTaskSettings.NAME, CohereEmbeddingsTaskSettings::new)
         );
+        namedWriteables.add(
```

Comment:
I think we need the service settings too right?

```
        namedWriteables.add(
            new NamedWriteableRegistry.Entry(ServiceSettings.class, CohereRerankServiceSettings.NAME, CohereRerankServiceSettings::new)
        );
```

Reply:
Thanks Jonathan! Do you think we can come up with a list of all the things that need to go in this file and add it as a comment here? I always have trouble figuring out what exactly needs to go in here.

Reply:
Yeah I believe it's any time we create a class that implements `NamedWriteable`. That kind of hard to know without following the `implements` of classes, but essentially anything that implements `TaskSettings`, `ServiceSettings`, or `InferenceServiceResults` at the moment.

Technically also `InferenceResults` but we shouldn't really be implementing that interface anymore within the inference plugin.

Reply:
Nice, I will add a comment specifying this at the top of this class.